### ISSUE TITLE
Support for type only imports & exports

### ISSUE THREAD

**Issue by chrstntdd** [alangpierce/sucrase#520]
Hello there! This project is fantastic and I use it heavily. Thank you for all your hard work!

I have a project using the latest stable TypeScript version (3.8.3) and tried to use [Type-Only Imports and Export](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) but was unable to compile with sucrase - throwing the following error.

```shell
SyntaxError: Error transforming src/index.ts: Unexpected token (8:13)
```

```ts
// src/index.ts
// ...
import type { TerserOpts } from "./types"
// ...
```

Are there any plans to support this feature in sucrase? If so, is there any way I can help contribute? Would love to have sucrase support this.
Cheers!

**Comment by alangpierce** [comment]
Hi @chrstntdd , thanks for the kind words. 😄

Yep, I've been following along on type-only imports, and certainly I want Sucrase to support them. I was hoping that they'd eventually get rid of the need to detect imported identifiers as type-only or not (which causes quite a bit of complexity and slowness in Sucrase), but looks like statements like `import {a, b} from 'foo';` still need to have `a` or `b` removed based on usage in the file.

From a code standpoint, getting this working will involve these steps:
1. To implement parsing, port all recent babel-parser changes, similar to https://github.com/alangpierce/sucrase/pull/487 . That should naturally pick up parsing the `type` token.
2. Change `CJSImportProcessor` and `ESMImportProcessor` to remove `import type` lines. I think this actually is already working because flow has the same syntax, so this just needs some TS tests I think.
3. Maybe change `CJSImportProcessor` and `ESMImportProcessor` to optionally no longer remove type-only import lines, equivalent to `importsNotUsedAsValues=preserve` in TS and [onlyRemoveTypeImports](https://babeljs.io/docs/en/babel-plugin-transform-typescript#onlyremovetypeimports) in Babel.

I'm actually unsure how to approach that last point, or whether to do anything at all, and I'd love feedback. I think my plan for my own projects is to use `importsNotUsedAsValues=error` everywhere so that I need to be explicit, and that means that the Sucrase behavior doesn't matter because I never intend to have type-only imports without `type`. Do you think you'd want a Sucrase flag to mimic `importsNotUsedAsValues=preserve` (keeping type-only import statements as side-effect-only imports)? Certainly if TS makes that the default, Sucrase will follow suit, but in general I'd prefer to keep Sucrase configuration minimal.

I think it probably makes sense for me to do step 1 since it's a bit hard to do without a lot of project context. But it would be great to get your feedback and if you could help test the feature when it's ready. At the moment, I don't have any codebases using that syntax that I can test with, so I worry that I may miss some detail.

**Comment by alangpierce** [comment]
Sorry for the delay! I just released this as Sucrase 3.14.0. Let me know if you see any issues!
